packages/service-library/src/servicelib/monitor_slow_callbacks.py

```diff
@@ -19,13 +21,22 @@ def enable(slow_duration_secs: float, incidents: List[SlowCallback]) -> None:
     _run = asyncio.events.Handle._run
 
     def instrumented(self):
-        t0 = time.monotonic()
-        retval = _run(self)
-        dt = time.monotonic() - t0
-        if dt >= slow_duration_secs:
-            task_info = _format_handle(self)
-            incidents.append(SlowCallback(msg=task_info, delay_secs=dt))
-            logger.warning("Executing %s took %.3f seconds", task_info, dt)
+        # unsetting profiler, helps with development mode and tests
+        sys.setprofile(None)
+
+        with Profiler(interval=slow_duration_secs) as profiler:
+            t0 = time.monotonic()
+
+            retval = _run(self)
+
+            dt = time.monotonic() - t0
+
+        profiler_result = profiler.output_text(unicode=True, color=False, show_all=True)
```

Comment:
is the indentation here correct? you are using the profile outside of the context manager no?

Reply:
and why doing it if the dt was not above the slow_duration_secs?

Reply:
> is the indentation here correct? you are using the profile outside of the context manager no?

Yes, indentation on `profiler_result` needs to be changed

Reply:
> and why doing it if the dt was not above the slow_duration_secs?

because the part which needs to be monitored is the async function not the profiler + the async function

Reply:
> is the indentation here correct? you are using the profile outside of the context manager no?

I had  for a moment the same impression but it seems managed context is to start/stop the profiler
```
profiler = Profiler()
profiler.start()

# code you want to profile

profiler.stop()

print(profiler.output_text(unicode=True, color=True))
```

Reply:
> is the indentation here correct? you are using the profile outside of the context manager no?

Actually it is correct. I've added a comment on why that is.